ledger/participant-integration-api/src/main/scala/platform/indexer/parallel/ParallelIndexerFactory.scala

```diff
@@ -115,16 +98,160 @@ object ParallelIndexerFactory {
             ).map(_ => ())
 
       def subscribe(readService: ReadService): Future[Source[Unit, NotUsed]] =
-        postgresDaoPool
-          .execute(_.initialize)
-          .map { case (offset, eventSeqId) =>
-            ingest(eventSeqId.getOrElse(0L))(readService.stateUpdates(beginAfter = offset))
-          }(scala.concurrent.ExecutionContext.global)
+        dbDispatcher
+          .executeSql(metrics.daml.parallelIndexer.initialization)(
+            storageBackend.initialize
+          ) // TODO here is an implicit LoggingContext injected from the factory apply implicit param, is that good this way? shouldn't we create here something for the purpose?
+          .map(initialized =>
+            ingest(initialized.lastEventSeqId.getOrElse(0L))(
+              readService.stateUpdates(beginAfter = initialized.lastOffset)
+            )
+          )(scala.concurrent.ExecutionContext.global)
 
       toIndexer(subscribe)(mat)
     }
   }
 
+  case class RunningBatch[T](
```

Comment:
This class has many properties relevant to different stages of the indexing pipeline. Not all properties are used in each stage, and it's not obvious what they mean. I think this class needs a good description of all members.

For example, what exactly is `lastSeqEventId`? It is assigned by the seqMapper, and seems to store the "maximum of the sequential event ids within this batch". Then the tailer takes the `lastSeqEventId` value of the last encountered batch, but that doesn't mean all sequential ids up to that point were written, right? The batches will not arrive at the tailer in the order of increasing sequential event ids.

Reply:
good point, will add descriptions